pytensor/tensor/rewriting/blockwise.py

```diff
@@ -234,6 +239,26 @@ def local_blockwise_reshape(fgraph, node):
         return [new_out]
 
 
+@register_stabilize
+@register_specialize
+@node_rewriter([Blockwise])
+def local_blockwise_of_subtensor(fgraph, node):
+    """Rewrite Blockwise of Subtensor, where the only batch dimensions are the inputs."""
+    if not isinstance(node.op.core_op, Subtensor):
```

Comment:
Didn't you tell me we can put `Blockwise(Subtensor)` in the decorator instead of always doing this check?

Reply:
Subtensor itself is parametrized by `idx_list`, I would need to create all parametrization of Blockwise(Subtensor), and track them (which is not possible since Subtensor can have infinite parametrizations (ignoring that numpy arrays can have atmost 64 dims).

You could for instance for Conv1d which only has two parametrizations. You would do:
```python
@node_rewrites([Blockwise(Conv1d(mode="full")), Blockwise(Conv1d(mode="valid"))])
```

Reply:
This opens the door to the idea of making rewrites more "database index-like". Stuff like `SequenceOf(Solve, Diagonal)`, that the `PatternNodeRewriter` tries to do, but fails for the very same kind of parametrized Ops.

Reply:
Opened issue: https://github.com/pymc-devs/pytensor/issues/1354